.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,125 @@
+name: End-to-End Tests
+on:
+  workflow_dispatch:
+  pull_request:
+concurrency: build-wsl
+jobs:
+  start_vm:
+    strategy:
+      max-parallel: 1
+    runs-on: ubuntu-latest
+    steps:
+      - uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_VM_CREDS }}
+      - name: Start the Runner
+        run: |
+          az vm start --name wsl-ci --resource-group wsl
+  rdp:

```

Comment:
We'll review RDP session requirement to run WSL from a session other than session 0 at a later stage.